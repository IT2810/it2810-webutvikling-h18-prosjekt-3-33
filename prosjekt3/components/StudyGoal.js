import React from 'react';
import {
  View, Text, StyleSheet, ScrollView, Button, TouchableOpacity

} from 'react-native';
import Colors from '../constants/Colors.js';
import PedometerCounter from './PedometerCounter';
import StudyCounter from './StudyCounter';

export default class StepGoal extends React.Component {
  state = {
    studyGoal: 0,
  }
  componentDidMount(){
    this.setState({studyGoal: this.props.studyGoal})
  }
  componentDidUpdate(prevProps, prevState){
    if(this.props.studyGoal != prevProps.studyGoal){
      this.setState({studyGoal: this.props.studyGoal})
    }
  }
  render() {
    return(
      <ScrollView style={styles.container}>
        <View style={styles.goalContainer}>
          <Text style={styles.goalTitle}>Number of hours studied today </Text>
          <StudyCounter studyGoal={this.props.studyGoal} />
        </View>
      </ScrollView>
    );
  }
}
const styles = StyleSheet.create({
  goalTitle: {
    fontSize: 16,
    color: 'white',
  },
  goalContainer: {
    backgroundColor: Colors.goalComponentOne,
    height: 150,
    padding: 10,
    marginTop: 20
  },
  container: {
    padding: 20
  },
  emptyContainer: {
    height: 80
  }

});
