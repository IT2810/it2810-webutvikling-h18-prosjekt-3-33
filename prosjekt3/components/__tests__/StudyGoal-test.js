import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import MobileButton from 'react-native';
import StudyGoal from '../StudyGoal';
import StudyCounter from '../StudyCounter';
import { Pedometer } from 'expo';


describe('StudyGoal', () => {

	it('renders correctly', () => {
	  	const studygoal = renderer.create(<StudyGoal />).toJSON();
	  	expect(studygoal).toMatchSnapshot();
	});

});

describe('StudyCounter', () => {
  it('renders correctly', () => {
    const studycounter = renderer.create(<StudyCounter />).toJSON();
    expect(studycounter).toMatchSnapshot();
  });

  it('increments counter correctly', () => {
    const studycounter = renderer.create(<StudyCounter />).getInstance();
    const prevCounter = studycounter.state.currentStudyCount;
    studycounter.incrementCounter(5);
   	const newCounter = studycounter.state.currentStudyCount;
    expect(newCounter).toEqual(prevCounter+1);
  });

	it('decrements counter correctly', () => {
    const studycounter = renderer.create(<StudyCounter />).getInstance();
    const prevCounter = studycounter.state.currentStudyCount;
    studycounter.decrementCounter(5);
   	const newCounter = studycounter.state.currentStudyCount;
    expect(newCounter).toEqual(prevCounter-1);
  });

	it('fetches from props', () => {
		const studyGoal= 5;
		const studycounter = renderer.create(<StudyCounter studyGoal={studyGoal} />).getInstance();
		const prevGoal = studycounter.state.studyGoal;
		expect(prevGoal).toEqual(studyGoal);
	});
	it('calculates progress colors', () => {
		let goal= 8;
		let progress = 8;
		const studycounter = renderer.create(<StudyCounter />).getInstance();
		const result = studycounter.progressBarStyle(progress,goal);
		let expectedResult = {
			flex: 1,
    	backgroundColor: '#75CDA8',
		}
		expect(result).toEqual(expectedResult);

	});
});
