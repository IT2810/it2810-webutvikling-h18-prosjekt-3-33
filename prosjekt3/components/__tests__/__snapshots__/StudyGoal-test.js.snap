// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudyCounter renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "marginTop": 15,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 300,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#c11537",
          "height": 40,
          "width": 40,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            -
          </Text>
        </View>
      </View>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      0
       out of 
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#15c129",
          "height": 40,
          "width": 40,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            +
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#eee",
        "flex": 1,
        "flexDirection": "row",
        "height": 20,
        "marginTop": 20,
        "width": 300,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#000",
          "flex": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`StudyGoal renders correctly 1`] = `null`;
